scipy/_lib/_array_api.py

```diff
@@ -196,15 +196,15 @@ def copy(x, *, xp=None):
 
 
 def is_numpy(xp):
-    return xp.__name__ == 'scipy._lib.array_api_compat.numpy'
+    return xp.__name__ in ('numpy', 'scipy._lib.array_api_compat.numpy')
 
 
 def is_cupy(xp):
-    return xp.__name__ == 'scipy._lib.array_api_compat.cupy'
+    return xp.__name__ in ('cupy', 'scipy._lib.array_api_compat.cupy')
 
 
 def is_torch(xp):
-    return xp.__name__ == 'scipy._lib.array_api_compat.torch'
+    return xp.__name__ in ('torch', 'scipy._lib.array_api_compat.torch')
```

Comment:
These were deliberate at the time, since NumPy (lacking `__array_namespace__`) would always return `array_api_compat.numpy` from `array_namespace`. But yes, I think it's good to change to this now (it used to be `'numpy' in xp.__name__`).